R/utils.R

```diff
@@ -80,18 +80,21 @@ link_arms <- function(
 #' The associated \code{string} comes from metadata outputs.
 #'
 #' @returns A tidy \code{tibble} from a matrix giving raw and label outputs to
-#' be used in later functions.
+#' be used in later functions if \code{return_vector = FALSE}, the default.
+#' Otherwise a vector result in a c(raw = label) format to use with
+#' dplyr::recode
 #'
 #' @param string A \code{db_metadata$select_choices_or_calculations} field
 #' pre-filtered for checkbox \code{field_type}
+#' @param return_vector logical for whether to return result as a vector
 #'
 #' @importFrom stringi stri_split_fixed
 #' @importFrom tibble as_tibble is_tibble
 #' @importFrom rlang .data
 #'
 #' @keywords internal
 
-parse_labels <- function(string) {
+parse_labels <- function(string, return_vector = FALSE) {
```

Comment:
Since this is an internal function I think we can safely assume that nobody depends on the old behavior? Do we need the `return = FALSE` code anywhere else? If not, I would scrap it.

Reply:
Yeah we use it here:
https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/2e445b99a036e036e02d2d8b0c3b2343eb645bbd/R/utils.R#L173-L183

which is used to add all the distinct checkbox option fields to the metadata. Could probably refactor that to use the vector output from `parse_labels()` rather than `tibble` but I'd also just as happily not touch it. What do you think?

Reply:
I think we can leave as is then!